proposals/numeric-intptr.md

```diff
@@ -0,0 +1,344 @@
+
+### 11.8.5 Bitwise complement operator
+
+The predefined bitwise complement operators are:
+
+```csharp
+...
+nint operator ~(nint x);
+nuint operator ~(nuint x);
+```
+
+### 11.8.6 Prefix increment and decrement operators
+
+Predefined `++` and `--` operators exist for the following types: `sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, **`nint`, `nuint`**, `long`, `ulong`, `char`, `float`, `double`, `decimal`, and any enum type.
+
+## 11.9 Arithmetic operators
+
+### 11.9.2 Multiplication operator
+
+The predefined multiplication operators are listed below. The operators all compute the product of `x` and `y`.
+
+- Integer multiplication:
+
+  ```csharp
+  ...
+  nint operator *(nint x, nint y);
+  nuint operator *(nuint x, nuint y);
+  ```
+
+### 11.9.3 Division operator
+
+The predefined division operators are listed below. The operators all compute the quotient of `x` and `y`.
+
+- Integer division:
+
+  ```csharp
+  ...
+  nint operator /(nint x, nint y);
+  nuint operator /(nuint x, nuint y);
+  ```
+
+### 11.9.4 Remainder operator
+
+The predefined remainder operators are listed below. The operators all compute the remainder of the division between `x` and `y`.
+
+- Integer remainder:
+
+  ```csharp
+  ...
+  nint operator %(nint x, nint y);
+  nuint operator %(nuint x, nuint y);
+  ```
+
+### 11.9.5 Addition operator
+
+- Integer addition:
+
+  ```csharp
+  ...
+  nint operator +(nint x, nint y);
+  nuint operator +(nuint x, nuint y);
+  ```
+
+### 11.9.6 Subtraction operator
+
+- Integer subtraction:
+
+  ```csharp
+  ...
+  nint operator –(nint x, nint y);
+  nuint operator –(nuint x, nuint y);
+  ```
+
+## 11.10 Shift operators
+
+The predefined shift operators are listed below.
+
+- Shift left:
+
+  ```csharp
+  ...
+  nint operator <<(nint x, int count);
+  nuint operator <<(nuint x, int count);
+  ```
+
+- Shift right:
+
+  ```csharp
+  ...
+  nint operator >>(nint x, nint count);
+  nuint operator >>(nuint x, nint count);
+  ```
+
+  The `>>` operator shifts `x` right by a number of bits computed as described below.
+
+  When `x` is of type `int`, **`nint`** or `long`, the low-order bits of `x` are discarded, the remaining bits are shifted right, and the high-order empty bit positions are set to zero if `x` is non-negative and set to one if `x` is negative.
+
+  When `x` is of type `uint`, **`nuint`** or `ulong`, the low-order bits of `x` are discarded, the remaining bits are shifted right, and the high-order empty bit positions are set to zero.
+
+- Unsigned shift right:
+
+  ```csharp
+  ...
+  nint operator >>>(nint x, nint count);
+  nuint operator >>>(nuint x, nint count);
+  ```
+
+For the predefined operators, the number of bits to shift is computed as follows:
+\[...]
+- When the type of `x` is `nint` or `nuint`, the shift count is given by the low-order five bits of `count` on a 32 bits platform, or the lower-order six bits of `count` on a 64 bits platform.  
```

Comment:
>32 bits [](http://example.com/codeflow?start=113&length=7)

Should these be "32 bit" and "64 bit" rather than "... bits" here and in the next sentence?